spec/pagerduty/events_api_v2_spec.rb

```diff
@@ -0,0 +1,353 @@
+          )
+        }
+        Then {
+          expect(transport).to have_received(:send_payload).with(
+            routing_key:  "a-test-routing-key",
+            event_action: "trigger",
+            dedup_key:    nil,
+            payload:      {
+              summary:        "summary",
+              source:         "source",
+              severity:       "critical",
+              timestamp:      "2015-07-17T08:42:58Z",
+              component:      "component",
+              group:          "group",
+              class:          "class",
+              custom_details: {
+                random: "random",
+              },
+            },
+            images:       [{
+              src:  "http://via.placeholder.com/350x150",
+              href: "https://example.com/",
+              alt:  "Example text",
+            }],
+            links:        [{
+              href: "https://example.com/",
+              text: "Link text",
+            }],
+            client:       "Sample Monitoring Service",
+            client_url:   "https://monitoring.example.com",
+          )
+        }
+      end
+
+      context "given simple event details" do
+        When { pagerduty.trigger(simple_incident_details) }
+        Then {
+          expect(transport).to have_received(:send_payload).with(
+            routing_key:  "a-test-routing-key",
+            dedup_key:    nil,
+            event_action: "trigger",
+            payload:      {
+              summary:  "summary",
+              source:   "source",
+              severity: "critical",
+            },
+          )
+        }
+      end
+
+      context "configured with an HTTP proxy" do
+        Given(:http_proxy) {
+          {
+            host:     "test-proxy-host",
+            port:     "test-proxy-port",
+            username: "test-proxy-username",
+            password: "test-proxy-password",
+          }
+        }
+        When(:incident) { pagerduty.trigger(simple_incident_details) }
+        Then {
+          expect(Pagerduty::HttpTransport).to have_received(:new).with(
+            a_hash_including(
+              proxy: {
+                host:     "test-proxy-host",
+                port:     "test-proxy-port",
+                username: "test-proxy-username",
+                password: "test-proxy-password",
+              },
+            ),
+          )
+        }
+      end
+    end
+
+    describe "handling responses" do
+      context "PagerDuty successfully creates the incident" do
+        Given {
+          allow(transport).to receive(:send_payload).and_return(
+            "status"    => "success",
+            "dedup_key" => "incident-key-in-response",
+            "message"   => "Event processed",
+          )
+        }
+        When(:incident) { pagerduty.trigger(simple_incident_details) }
+        Then { expect(incident).to be_a Pagerduty::EventsApiV2::Incident }
+        Then { incident.incident_key == "incident-key-in-response" }
```

Comment:
Should this be an expectation?

```suggestion
        Then { expect(incident.incident_key).to eq "incident-key-in-response" }
```

Reply:
This is actually the standard way to use `Then` blocks in [rspec-given](https://github.com/rspec-given/rspec-given). As I understand, the block should always return `true`. There is an explanation in the [`Then` documentation](https://github.com/rspec-given/rspec-given#then)

In this instance, if I change code under test to not set the `incident_key` this test fails with the following output:

```
Failures:

  1) Pagerduty#trigger handling responses PagerDuty successfully creates the incident         Then { incident.incident_key == "incident-key-in-response" }

     Failure/Error: Then { incident.incident_key == "incident-key-in-response" }
     
       Then expression failed at /Users/orien/src/pagerduty/spec/pagerduty/events_api_v2_spec.rb:169
       expected: nil
       to equal: "incident-key-in-response"
         false   <- incident.incident_key == "incident-key-in-response"
         nil     <- incident.incident_key
         #<Pagerduty::EventsApiV2::Incident:0x00007fbf20a7c8a8 @integration_key="a-test-routing-key", @incident_key=nil, @transport=#<InstanceDouble(Pagerduty::HttpTransport) (anonymous)>>
                 <- incident
     # ./spec/pagerduty/events_api_v2_spec.rb:169:in `block in Then'
```

It's pretty awesome how it breaks down the comparison so you can see different values in the expression. Pretty handy for debugging.

Reply:
TIL.